src/App.js

```diff
@@ -1,34 +1,56 @@
 import './App.css';
 import {Checklist} from './Checklist.js'
 import {AddTaskPage} from "./AddTaskPage";
-import React, {useEffect, useState} from "react";
+import React, {useState} from "react";
 import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
 import {ModifyTaskPage} from "./ModifyTaskPage";
 import {useCollection} from "react-firebase-hooks/firestore";
 import firebase from "firebase/compat";
+import {AddNewListPage} from "./AddNewListPage";
+import {ConfirmDeletePage} from "./ConfirmDeletePage";
 
 export function App(props) {
+
     const [sortValue, setSortValue] = useState("title");
-    const [value, loading, error] = useCollection(props.collection.orderBy(sortValue, "asc"));
+    const [listValue, loading, error] = useCollection(props.collection);
+    let lists = [
+        // {
+        //     title: "other list",
+        //     id: "v1-1636655036490-9127893358184"
+        // }
+        {
+            title: "To Do",
+            id: "v1-1636654768436-2119059561616"
+        }
+    ];
+    if (listValue !== undefined) {
+        console.log("reading in lists");
+        lists = listValue.docs.map(doc =>
+            doc.data());
+    }
+    const [currentList, setCurrentList] = useState(lists[0].id);
+    const [taskValue, taskLoading, taskError] = useCollection(props.collection.doc(currentList).collection("tasks").orderBy(sortValue, "asc"));
```

Comment:
Might be easier to manage if you had the App be simpler, with no interaction with Firestore:
*  turn the Select a List into its own component (that is responsible for fetching the data from Firestore, adding, deleting, renaming, etc.).  It just reports back to the App what list is selected (if any).
* Then, have the Checklist be responsible for getting its subcollection from Firestore. It could just be told its list ID. It could then handle all the interaction with Firestore (adding/deleting/editing/retrieving tasks).